scripts/zkai_native_attention_mlp_adapter_air_frontier_gate.py

```
@@ -0,0 +1,646 @@
+        if _int(row.get("index"), f"row {index} index") != index:
+            raise AdapterAirFrontierError("projection row index drift")
+        primary_source = _int(row.get("primary_source_index"), f"row {index} primary source")
+        mix_source = _int(row.get("mix_source_index"), f"row {index} mix source")
+        primary = _int(row.get("primary_q8"), f"row {index} primary")
+        mix = _int(row.get("mix_q8"), f"row {index} mix")
+        bias = _int(row.get("bias_q8"), f"row {index} bias")
+        numerator = _int(row.get("numerator_q8"), f"row {index} numerator")
+        output = _int(row.get("output_q8"), f"row {index} output")
+        denominator = _int(row.get("denominator"), f"row {index} denominator")
+        if primary_source != index % ATTENTION_FLAT_CELLS:
+            raise AdapterAirFrontierError("primary source policy drift")
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)